src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/publishers/v2/TransactionExpiredEventPublisher.kt

```diff
@@ -100,21 +103,35 @@ class TransactionExpiredEventPublisher(
         toEvent(transaction)
             .flatMap { eventStoreRepository.save(it) }
             .flatMap { event ->
-                viewRepository
-                    .findByTransactionId(transaction.transactionId.value())
-                    .cast(TransactionV2::class.java)
-                    .flatMap {
-                        it.status = newStatus
-                        viewRepository.save(it)
-                    }
-                    .flatMap { Mono.just(event) }
+                conditionallySaveTransactionView(transaction, newStatus).then(Mono.just(event))
             }
 
-    override fun toEvent(baseTrasaction: BaseTransactionV2): Mono<TransactionExpiredEventV2> =
+    /**
+     * Save the transaction in the transactions-view collection, iff the transactions-view update is
+     * enabled. Otherwise, do nothing.
+     */
+    private fun conditionallySaveTransactionView(
+        transaction: BaseTransactionV2,
+        newStatus: TransactionStatusDto
+    ): Mono<Unit> =
+        if (transactionsViewUpdateEnabled) {
+            viewRepository
+                .findByTransactionId(transaction.transactionId.value())
+                .cast(TransactionV2::class.java)
+                .flatMap {
+                    it.status = newStatus
+                    viewRepository.save(it)
+                }
+                .then(mono {})
+        } else {
+            Mono.empty()
+        }
```

Comment:
What about this refactoring?
```suggestion
    private fun conditionallySaveTransactionView(
        transaction: BaseTransactionV2,
        newStatus: TransactionStatusDto
    ): Mono<TransactionV2> =
        Mono.just(transactionsViewUpdateEnabled)
            .filter { it }
            .map { viewRepository.findByTransactionId(transaction.transactionId.value()) }
            .flatMap { it.cast(TransactionV2::class.java) }
            .flatMap { 
                it.status = newStatus
                viewRepository.save(it)
            }
```

Reply:
Much more elegant and _reactive-way_, thanks. Refactored in [here](https://github.com/pagopa/pagopa-ecommerce-transactions-scheduler-service/pull/193/commits/4fb677218d26bcb43a55bde8982071f4e1b23889).